EchoPro/survey.py

```diff
@@ -36,741 +23,440 @@ class Survey:
-        class variable ``params``.
-
-        Notes
-        -----
-        The class variable ``params`` will be directly modified.
-        """
-
-        # convert the root directory to a Path object
-        self.params["data_root_dir"] = Path(self.params["data_root_dir"])
-
-        for param_name, param_val in self.params.items():
-
-            # convert each filename path to a Path object
-            if "filename" in param_name:
-                self.params[param_name] = Path(param_val)
-
-    def load_survey_data(self, file_type: str = "all") -> None:
+    def load_configuration( init_config_path: Path , 
+                            survey_year_config_path: Path ):
         """
         Loads the biological, NASC, and stratification
         data using parameters obtained from the configuration
         files.
 
         Parameters
         ----------
-        file_type : str
-            Specifies what survey data should be loaded.
-            Possible options:
-
-            - 'all' -> loads all survey data
-            - 'biological' -> only loads the biological data
-            - 'strata' -> only loads the stratification data
-            - 'nasc' -> only loads the NASC data
+        self : EchoPro.Survey
+        init_config_path : pathlib.Path
+            A string specifying the path to the initialization YAML file
+        survey_year_config_path : pathlib.Path
+            A string specifying the path to the survey year YAML file
 
         Notes
         -----
-        This function assigns class variables obtained from loading the
-        data. Specifically, the following class variables are created
-        for the file_type:
-
-        - ``file_type='biological'``
-            - ``self.length_df``
-            - ``self.specimen_df``
-
-        - ``file_type='strata'``
-            - ``self.strata_df``
-            - ``self.geo_strata_df``
-            - ``self.strata_sig_b``
-
-        - ``file_type='nasc'``
-            - ``self.nasc_df``
-        """
-
-        if file_type not in ["all", "biological", "strata", "nasc"]:
-            raise ValueError(
-                "file_type must be 'all', 'biological', 'strata', or 'nasc'!"
+        This function parses the configuration files and incorporates them into
+        the Survey class object. This initializes the `config` attribute that 
+        becomes available for future reference and functions.
+        """
+        ### Validate configuration files
+        # Retreive the module directory to begin mapping the configuration file location
+        current_directory = os.path.dirname(os.path.abspath(__file__))
+
+        # Build the full configuration file paths and verify they exist
+        config_files = [init_config_path, survey_year_config_path]
+        config_paths = [ Path(current_directory + '/../config_files/' + file) for file in config_files ]
+        config_existence = [path.exists() for path in config_paths]
```

Comment:
The input paths should be completely independent from the module directory, since they should be either relative path to the current directory of wherever the user is, or absolute path. `Path` handles both.

Below is my suggestion:
```suggestion
        # Build the full configuration file paths and verify they exist
        config_existence = [init_config_path.exists(), survey_year_config_path.exists()]
```

Reply:
Ah I see. Previously, I was just using the filename itself as the input (e.g. `initialization_config.yml`) since I assumed that this would not change in terms of the directory location within `EchoPro`. I am fine with maintaining the same input structures as shown in the `echopro_workflow.ipynb` and `transect_selection_workflow.ipynb` example notebooks.

Reply:
This appears to now work with the absolute path (e.g. `'C:/Users/Brandyn/Documents/GitHub/EchoPro/config_files/initialization_config.yml'`) and relative path (e.g. `./config_files/initialization_config.yml`), so I will resolve this conversation.